sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JsonFileFormat.scala

```diff
@@ -114,7 +114,16 @@ class JsonFileFormat extends TextBasedFileFormat with DataSourceRegister {
     }
 
     (file: PartitionedFile) => {
-      val parser = new JacksonParser(actualSchema, parsedOptions)
+      // SPARK-21610: when the `requiredSchema` only contains `_corrupt_record`,
```

Comment:
Yeah, we've also considered this possibility. But currently we thought a query like `dfFromFile.select($"_corrupt_record")` shows all nulls looks countering intuition because the query is intended to get the corrupt records among the JSON records. That is why we provide a fix like current one. IMO, it's meaningless to have all nulls for `_corrupted_record`.